modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/MultiEncodeWorkflowOperationHandler.java

```diff
@@ -0,0 +1,545 @@
+   *
+   * @throws NotFoundException if the workspace doesn't contain the requested file
+   */
+  private WorkflowOperationResult multiencode(MediaPackage src, WorkflowOperationInstance operation)
+          throws EncoderException, IOException, NotFoundException, MediaPackageException, WorkflowOperationException {
+    MediaPackage mediaPackage = (MediaPackage) src.clone();
+    // Check which tags have been configured
+    String[] sourceTags = getConfigAsArray(operation, "source-tags");
+    String[] sourceFlavors = getConfigAsArray(operation, "source-flavors");
+    String[] targetTags = getConfigAsArray(operation, "target-tags");
+    String[] targetFlavors = getConfigAsArray(operation, "target-flavors");
+    String tagWithProfileConfig = StringUtils.trimToNull(operation.getConfiguration("tag-with-profile"));
+    boolean tagWithProfile = tagWithProfileConfig != null && Boolean.parseBoolean(tagWithProfileConfig);
+
+    // Make sure either one of tags or flavors are provided
+    if (sourceFlavors == null && sourceTags == null) {
+      logger.info("No source tags or flavors have been specified, not matching anything");
+      return createResult(mediaPackage, Action.CONTINUE);
+    }
+    String[] profiles = getConfigAsArray(operation, "encoding-profiles");
+    if (profiles == null)
+      throw new WorkflowOperationException("Missing encoding profiles");
+
+    // Sort out the combinatorics of all the tags and flavors
+    List<ElementProfileTagFlavor> selectors = getSrcSelector(sourceFlavors, sourceTags, targetFlavors, targetTags,
+            profiles);
+
+    long totalTimeInQueue = 0;
+    Map<Job, JobInformation> encodingJobs = new HashMap<Job, JobInformation>();
+    // Find the encoding profiles - should only be one per flavor or tag
+    for (ElementProfileTagFlavor eptf : selectors) {
+      // Look for elements matching the tag and flavor
+      Collection<Track> elements = eptf.elementSelector.select(mediaPackage, true);
+      for (Track track : elements) { // For each source
+        logger.info("Encoding track {} using encoding profile '{}'", track, eptf.getProfiles().get(0).toString());
+        // Start encoding and wait for the result
+        encodingJobs.put(composerService.multiEncode(track, eptf.getProfiles()),
+                new JobInformation(track, eptf, tagWithProfile));
+      }
+    }
+
+    if (encodingJobs.isEmpty()) {
+      logger.info("No matching tracks found");
+      return createResult(mediaPackage, Action.CONTINUE);
+    }
+
+    // Wait for the jobs to return
+    if (!waitForStatus(encodingJobs.keySet().toArray(new Job[encodingJobs.size()])).isSuccess()) {
+      throw new WorkflowOperationException("One of the encoding jobs did not complete successfully");
+    }
+
+    // Process the result
+    for (Map.Entry<Job, JobInformation> entry : encodingJobs.entrySet()) {
+      Job job = entry.getKey();
+      Track track = entry.getValue().getTrack(); // source
```

Comment:
Maybe chose a variable name like sourceTrack instead of commenting the meaning of the variable...